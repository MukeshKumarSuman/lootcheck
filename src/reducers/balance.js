import * as constants from '../actions/constants';

const balance = (state = 0, action) => {
    switch(action.type) {
        case constants.SET_BALANCE:
            return action.balance;
        case constants.DEPOSIT:
            return state + action.deposit;
        case constants.WIDTHDRAW:
            return state - action.widthdrawal;
        default:
            return state; 
    }

}
export default balance;
