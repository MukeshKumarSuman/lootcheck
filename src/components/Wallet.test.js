import React from 'react';
import { shallow } from 'enzyme';
import { Wallet } from "./wallet";
import balance from '../reducers/balance';

describe(' Wallet Module', () => {
    const mockDeposit = jest.fn();
    const mockWidthdraw = jest.fn();
    const props = {balance: 20, deposit: mockDeposit, widthdraw: mockWidthdraw}
    const wallet = shallow( <Wallet {...props}/>);

    it('render properly', () => {
        expect(wallet).toMatchSnapshot();
    });

    it('display the balance from props', () => {
        expect(wallet.find('.balance').text()).toEqual('Wallet balance: 20');
    });

    it('creates an inputto deposit into or widthdraw from the balance ', () => {
        expect(wallet.find('.input-wallet').exists()).toBe(true);
    });

    describe('when user types into the wallet input', () => {
        const userBalance = '25';

        beforeEach( () => {
            wallet.find('.input-wallet').simulate('change', {target: {value: userBalance}});
        });

        it('update the local wallet balance in state and converts it to number', () => {
            expect(wallet.state().balance).toEqual(parseInt(userBalance, 10));
        });

        describe('and the user wants to make a deposit', () => {

            beforeEach(() => {
                wallet.find('.btn-deposit').simulate('click');
            });

            it('dispatches the deposit() it receives from props with local balance', () => {
                expect(mockDeposit).toHaveBeenCalledWith(parseInt(userBalance, 10));
            });
        });

        describe('and the user wants to make a widthdrawal', () => {

            beforeEach(() => {
                wallet.find('.btn-widthdraw').simulate('click');
            });

            it('dispatches the widthdraw() it receives from props with local balance', () => {
                expect(mockWidthdraw).toHaveBeenCalledWith(parseInt(userBalance, 10));
            });
        });
    });
});
