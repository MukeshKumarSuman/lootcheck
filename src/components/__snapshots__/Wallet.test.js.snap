// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` Wallet Module render properly 1`] = `ShallowWrapper {}`;
